### PR TITLE
BeginnerTriathlete service integration

### PR DESCRIPTION
**Pull request by mattsummers**
I will contact you off-site with an API key. Very much looking forward to your critique and feedback. Thanks.

### CODE REVIEW

File: tapiriik/services/BeginnerTriathlete/beginnertriathlete.py [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+                raise APIExcludeActivity("Parse error " + deviceUploadFile + " " + str(e),
+                                         user_exception=UserException(UserExceptionType.Corrupt),
+                                         permanent=True)
+
+        return activity
+
+    def UploadActivity(self, serviceRecord, activity):
+        # We could try to recreate the activity using the user facing event entry methods, but if we can just upload the
+        # workout as a TCX, that seems even better.
+        # Would a .FIT file be better? This endpoint will support both file types.
```
Comment (cpfair) [inline tapiriik/services/BeginnerTriathlete/beginnertriathlete.py]: Yes, FIT has better support for non-GPS activities and summary statistics.

File: tapiriik/services/BeginnerTriathlete/beginnertriathlete.py [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+            "Invalid login - Unknown error",
+            block=True,
+            user_exception=UserException(UserExceptionType.Authorization, intervention_required=True))
+
+    # Get an activity summary over a date range
+    def DownloadActivityList(self, serviceRecord, exhaustive=False):
+        activities = []
+
+        if exhaustive:
+            # "Exhaustive" currently goes back one year
```
Comment (cpfair) [inline tapiriik/services/BeginnerTriathlete/beginnertriathlete.py]: Is there a way to actually fetch all activities? As it is, activities more than a year ago on a user's BT account will get duplicated.

Comment (mattsummers) [inline tapiriik/services/BeginnerTriathlete/beginnertriathlete.py]: Yes. I have implemented this change. An exhaustive sync will now go back as far as is necessary to grab all activities. Thanks.

File: tapiriik/services/BeginnerTriathlete/beginnertriathlete.py (hunk 1 above)
Comment (mattsummers) [inline tapiriik/services/BeginnerTriathlete/beginnertriathlete.py]: Very good. I am finalizing testing of this now.

Comment (mattsummers) [inline tapiriik/services/BeginnerTriathlete/beginnertriathlete.py]: Done. The uploads will be sent as FIT

**Comment by mattsummers** [comment]
Added support for rollbacks and a few other cleanups

**Pull request merged**